test/e2e/suite/certificates/additionaloutputformats.go

```diff
@@ -44,17 +47,20 @@ import (
 // events.
 var _ = framework.CertManagerDescribe("Certificate AdditionalCertificateOutputFormats", func() {
 	const (
-		issuerName = "certificate-additional-output-formats"
-		secretName = "test-additional-output-formats"
+		issuerName   = "certificate-additional-output-formats"
+		secretName   = "test-additional-output-formats"
+		fieldManager = "e2-test-field-manager"
 	)
 
 	createCertificate := func(f *framework.Framework, aof []cmapi.CertificateAdditionalOutputFormat) *cmapi.Certificate {
 		framework.RequireFeatureGate(f, utilfeature.DefaultFeatureGate, feature.AdditionalCertificateOutputFormats)
+		// This should be a sufficiently unique and descriptive name
+		certName := fmt.Sprintf("test-additional-output-formats-%d", time.Now().Unix())
 
 		crt := &cmapi.Certificate{
 			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "test-additional-output-formats-",
-				Namespace:    f.Namespace.Name,
+				Name:      certName,
+				Namespace: f.Namespace.Name,
```

Comment:
Why have you changed this? I thought that the `Apply` function below would return the `crt` with the `Name` field generated by the API server.

Reply:
As I understand, `GenerateName` cannot be used with server side apply. I actually don't find any documentation around it, I assume it's to do with where the name gets generated and that Patch really specifies just a patch to be applied to either an existing object or one-to-be created and that `GenerateName` can only be used with newly created resources, but I am not sure. Might look into it and ask a question upstream.

You can see the error that kubectl would throw if  `metadata.GenerateName` is used with server-side apply [here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/apply/apply.go#L480) and can observe it if you `kubectl apply --server-side=true -f <some-resource-config-with-generate-name-directive>`. I observed a 'name cannot be empty' error when trying to use `GenerateName` when applying this resource with the versioned client.

Reply:
Oh, I did not realise that! I am a "Doubting Thomas" so had to go and check for myself.

```
$ kubectl apply -f internal/cmd/manifests/cm1.yaml 
error: from foo-bar-: cannot use generate name with apply

$ go run ./
foo
2022/04/29 12:13:45 resource name may not be empty
exit status 1

``

Reply:
So I need to make a mental note to stop using GenerateName in future.